eZ/Publish/Core/REST/Server/Input/Parser/Query.php

```diff
@@ -104,4 +100,27 @@ private function processCriteriaArray(array $criteriaArray, ParsingDispatcher $p
 
         return (count($criteria) === 1) ? $criteria[0] : new CriterionValue\LogicalAnd($criteria);
     }
+
+    /**
+     * Handles SortClause data
+     * @param array $sortClasuesArray
+     * @param ParsingDispatcher $parsingDispatcher
+     *
+     * @return array
+     */
+    private function processSortClauses(array $sortClasuesArray, ParsingDispatcher $parsingDispatcher) {
+        $sortClauses = [];
+
+        foreach ($sortClasuesArray as $sortClauseName => $sortClauseData) {
+            if (is_array($sortClauseData) && isset($sortClauseData[0])) {
+                foreach ($sortClauseData as $singleSortClauseData) {
+                    $sortClauses[] = $this->dispatchSortClause($sortClauseName, $singleSortClauseData, $parsingDispatcher);
+                }
+            } else {
+                $sortClauses[] = $this->dispatchSortClause($sortClauseName, $sortClauseData, $parsingDispatcher);
```

Comment:
How about normalizing this case to an array and then always do the above foreach.